### ISSUE TITLE
noMisplacedAssertion: `test.prop(...)` from `@fast-check/vitest` is not recognised as a test function

### ISSUE THREAD

**Issue by CarlAllenn** [biomejs/biome#11454]
## Environment

Biome 2.5.7 (aqua-installed binary, macOS arm64), CLI, no LSP.

## Summary

`suspicious/noMisplacedAssertion` does not recognise `test.prop(...)`
from [`@fast-check/vitest`](https://github.com/dubzzz/fast-check/tree/main/packages/vitest)
as a test function, so it reads the property body as loose code and
reports every `expect` inside it.

This is the same class as the chained-call cases already fixed —
#2443 (`it.each`), #2547, #10635 (`{test,describe}.concurrent.*`),
fixed in #10680 (`test.each`) — with `.prop` being the member the
list does not yet carry.

## Reproduction

`biome.json`:

```json
{
  "linter": {
    "rules": {
      "preset": "none",
      "suspicious": { "noMisplacedAssertion": "on" }
    }
  }
}
```

`a.test.ts`:

```ts
import { fc, test } from "@fast-check/vitest";
import { expect } from "vitest";

test.prop([fc.string()])("round-trips", (s) => {
  expect(s).toBe(s);
});
```

`biome lint a.test.ts`:

```
a.test.ts:5:3 lint/suspicious/noMisplacedAssertion
```

## Control, same run

`test.each` — the shape #10680 fixed, and a documented `Valid` example
in `biome explain noMisplacedAssertion` — is clean in the same
invocation:

```ts
import { expect, test } from "vitest";

test.each([1, 2])("each is recognised", (n) => {
  expect(n).toBeGreaterThan(0);
});
```

So the difference is the member name, not the chained-call shape.

## Expected result

`expect` inside a `test.prop(...)` / `it.prop(...)` callback is inside a
test and should not be reported — `.prop` handled alongside `.each`,
`.concurrent`, `.skip` and the rest.

## Why this one matters more than an ordinary false positive

The rule guards a real class: an `expect` genuinely stranded outside
`it`/`test` reports nothing and the suite still passes. On a real
TypeScript tree we measured 8 findings from this rule, 2 of them this
`test.prop` case; because there is no way to keep the rule and exempt
the library, the practical choices are per-site suppressions or turning
off a correctness rule. We are keeping the rule on and suppressing per
site.

Same library as #11453, which is a separate defect in
`useConsistentTestIt`. Two rules, one unrecognised test API.

**Comment by biome-cookie** [comment]
Hello @CarlAllenn, please provide a minimal reproduction. You can use one of the following options:

- Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
- Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`

Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

**Comment by CarlAllenn** [comment]
Reproduction, as requested.

Playground (rule `noMisplacedAssertion`, test domain `all`): https://biomejs.dev/playground/?lintRules=noMisplacedAssertion&pane=Diagnostics&language=ts&ruleDomains.test=all&code=aQBtAHAAbwByAHQAIAB7ACAAZgBjACwAIAB0AGUAcwB0ACAAfQAgAGYAcgBvAG0AIAAiAEAAZgBhAHMAdAAtAGMAaABlAGMAawAvAHYAaQB0AGUAcwB0ACIAOwAKAGkAbQBwAG8AcgB0ACAAewAgAGUAeABwAGUAYwB0ACAAfQAgAGYAcgBvAG0AIAAiAHYAaQB0AGUAcwB0ACIAOwAKAHQAZQBzAHQALgBwAHIAbwBwACgAWwBmAGMALgBzAHQAcgBpAG4AZwAoACkAXQApACgAIgByAG8AdQBuAGQALQB0AHIAaQBwAHMAIgAsACAAKABzACkAIAA9AD4AIAB7AAoAIAAgAGUAeABwAGUAYwB0ACgAcwApAC4AdABvAEIAZQAoAHMAKQA7AAoAfQApADsACgB0AGUAcwB0AC4AZQBhAGMAaAAoAFsAMQAsACAAMgBdACkAKAAiAGUAYQBjAGgAIABpAHMAIAByAGUAYwBvAGcAbgBpAHMAZQBkACIALAAgACgAbgApACAAPQA%2BACAAewAKACAAIABlAHgAcABlAGMAdAAoAG4AKQAuAHQAbwBCAGUARwByAGUAYQB0AGUAcgBUAGgAYQBuACgAMAApADsACgB9ACkAOwA%3D

The file holds the case and its control together: the `expect` inside `test.prop(...)` is reported, the identical `expect` inside `test.each(...)` two lines below is not.

Same thing on the CLI with Biome 2.5.7, `{ "linter": { "domains": { "test": "all" }, "rules": { "recommended": false, "suspicious": { "noMisplacedAssertion": "on" } } } }`:

```
a.test.ts:5:3 lint/suspicious/noMisplacedAssertion ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! The assertion isn't inside a it(), test() or Deno.test() function call.

    4 │ test.prop([fc.string()])("round-trips", (s) => {
  > 5 │   expect(s).toBe(s);
      │   ^^^^^^
    6 │ });

Checked 2 files in 1446µs. No fixes applied.
Found 1 warning.
```

(the second file being the `test.each` control, clean in the same run).

Expected: `expect` inside a `test.prop(...)` / `it.prop(...)` callback is inside a test and is not reported, with `.prop` handled alongside `.each` and `.concurrent`. Actual: only the `.prop` member is unrecognised, so every assertion in the callback is reported.
